bazel/linux/defs.bzl

```diff
@@ -135,17 +135,74 @@ KernelModulesInfo = provider(
     },
 )
 
+def _import_kernel_bundle_dep(name, ki, d, inputs, extra_symbols):
+    """Extracts information necessary to import a kernel bundle.
+
+    A kernel bundle represents a single kernel module compiled for
+    multiple kernels. Given a bundle d, a rule named name, trying
+    to compile a module for kernel ki, updates inputs and extra_symbols
+    with the correct list of dependencies.
+
+    Args:
+      name: string, name of the target.
+      ki: KernelTreeInfo provider, describing the target kernel this
+          module is being built for.
+      d: a Target object, obtained from a label or label_list attribute,
+         representing a dependency to link with this module.
+      inputs: list of File objects, updated with additional inputs
+         required by this dependency.
+      extra_symbol: list of File objects, updated with additional
+         Module.symvers files. 
+
+    Returns:
+      Updates inputs and extra_symbols, returns None.
+    """
+
+    matched = 0
+    for sub in d[KernelBundleInfo].modules:
+        matched += int(_import_kernel_modules_dep(name, ki, sub, inputs, extra_symbols))
+
+    if matched <= 0:
+        fail("%s is being built for kernel %s, depends on target %s, but this " +
+             "target is NOT being built for the specified kernel", name, ki.package, d.label.name)
+
```

Comment:
Will this fail early, or will it fail only when building `name`, after a dep target has been built for all the other kernels?